cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/LookupTableConfig.java

```diff
@@ -32,41 +34,39 @@
     DATASET
   }
 
-  /**
-   * When type is DATASET, {@link #properties} is interpreted as {@link DatasetProperties}.
-   */
   private final TableType type;
-  private final Map<String, Object> properties;
 
-  public LookupTableConfig(TableType type, Map<String, Object> properties) {
+  @Nullable
+  private final Map<String, String> datasetProperties;
+
+  @Nullable
+  private final CacheConfig cache;
+
+  public LookupTableConfig(TableType type, @Nullable CacheConfig cache,
+                           @Nullable Map<String, String> datasetProperties) {
```

Comment:
Why accept nullable? In general, having method that require user pass in `null` is not good. Better have a overloading constructor that doesn't that datasetProperties and default it to an empty map.